packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeLoader.ts

```diff
@@ -46,25 +48,29 @@ export class HierarchyLoader implements IHierarchyLoader {
     parent: TreeModelHierarchyNode<ParentHierarchyNode> | TreeModelRootNode,
     getInstanceFilter: (node: TreeModelRootNode | TreeModelHierarchyNode<ParentHierarchyNode>) => GenericInstanceFilter | undefined,
     buildNode?: (node: TreeModelHierarchyNode) => TreeModelHierarchyNode,
+    ignoreCache?: boolean,
   ) {
     return defer(async () =>
       this._hierarchyProvider.getNodes({
         parentNode: parent.nodeData,
         hierarchyLevelSizeLimit: parent.hierarchyLimit,
         instanceFilter: getInstanceFilter(parent),
+        ignoreCache,
       }),
     ).pipe(
       catchError((err) => {
+        const node: TreeModelInfoNode = {
+          id: `${parent.id ?? ""}-${err.message}`,
+          parentId: parent.id,
+          type: "Unknown",
+          message: "Failed to create hierarchy level",
+        };
         if (err instanceof RowsLimitExceededError) {
-          const node: TreeModelInfoNode = {
-            id: `${parent.id ?? ""}-${err.message}`,
-            parentId: parent.id,
-            type: "ResultSetTooLarge",
-            message: err.message,
-          };
+          node.type = "ResultSetTooLarge";
+          node.message = err.message;
           return of([node]);
         }
-        throw err;
+        return of([node]);
```

Comment:
Creating an object and overwriting its attributes right away doesn't make much sense, IMO. If you don't want to repeat the same attributes, I suggest doing the following:
```ts
const nodeProps = {
  id: ...,,
  parentId: ...,
};
if (x) {
  return of([{ ...nodeProps, type: "", message: "" }]);
}
return of([{ ...nodeProsp, type: "", message: "" }]);
```